### PR TITLE
Feature/redirect unit test logging

### PR DESCRIPTION
**Pull request by jweiser**
Excessive log output was being produced while running unit tests.  This PR adds configuration that turns off logging for tests only.  Running this locally, "mvn test" produces quieter output for successful tests but still outputs when, and why, a test fails.

### CODE REVIEW

**Comment by SolomonShorser-OICR** [comment]
What is "excessive"? Was it a specific class or package? If so, there are ways to configure logging at the class/package level from the config file. This is common with some 3rd-party packages (apache HTTPClient, I'm looking at YOU!). If the application _itself_ is too noisy, that might indicate the developers have added too many logging statements or haven't chosen appropriate levels for different kinds of logging statements. 

It looks like you've turned off ALL logging. That seems like a good way to frustrate/confuse future testers. I'd either look into tweaking the logging config to output less for some noisy packages, _or_ fix the logging statements in the application itself.

**Comment by jweiser** [comment]
It turns off logging for the tests only.  The main program logging is unaffected.

The `logging.properties` file with the line ` java.util.logging.ConsoleHandler.level=OFF` was to silence logging done by the Neo4J driver.  There wasn't really another way to get at it and it doesn't affect logging used in the main program since that is done by log4j.

The `log4j2.xml` in "src/test/resources" file silences log4j statements but only in testing.  I can look into customizing this more, but for this project, I think it makes sense to keep it off.  We're really only interested in details when a test fails and the Hamcrest asserts give a good indication why this happens when it happens.

**Comment by SolomonShorser-OICR** [comment]
There are other ways to silence neo4j (and Apache HTTP client, if that does it too). This may involve adding specific loggers for "noisy" packages. I'll look for some links and add them here...

**Comment by SolomonShorser-OICR** [comment]
Ok this is a bit ugly, but here's one way I got it working for the docker project: 

https://github.com/reactome/container/blob/92b8f95687187166c47ba2b0a2b8c6aae0ff7bc8/stand-alone-content-service/content-service.dockerfile#L16-L17

This example shows altering the log4j properties file _and_ the log4j XML config. Might not be necessary to do both, depending on your situation, but I still think this is preferable to silencing _all_ logging.

**Comment by jweiser** [comment]
Okay, the latest commits:

- sets the minimal logging output from the Jersey Server used by the Neo4J Harness test package (only used in unit testing) to be WARNING (i.e. only warnings and higher).
- redirect the log files from the main program to src/test/resources/logs during unit testing so they are available but don't obscure the output from "mvn test" (the amount and location of logging is configurable from "src/test/resources/log4j2.xml"

File: src/test/resources/log4j2.xml [hunk 1]
```diff
@@ -1,13 +1,63 @@
...
 		<Console name="Console" target="SYSTEM_OUT">
-			<PatternLayout pattern="%d{YYYY-MM-dd HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n" />
+			<PatternLayout pattern="${logPattern}" />
 		</Console>
+
+		<RollingFile
+			name="mainLogFile"
+			fileName="${baseDir}/Main-${date:MM-dd-yyyy_HH.mm.ss}.log"
+			filePattern="${baseDir}/Main-%d{MM-dd-yyyy_HH.mm.ss}.log"
+		>
```
Comment (SolomonShorser-OICR) [inline src/test/resources/log4j2.xml]: Just curious: is IntelliJ doing this formatting?

Comment (jweiser) [reply src/test/resources/log4j2.xml]: No, I've been formatting this way.  It's a convention that I've seen used in many places when you have several attributes with long values, they are moved to one on each line.

Comment (SolomonShorser-OICR) [reply src/test/resources/log4j2.xml]: Huh. The ">" dangling on its own line is a little weird looking to me. I don't think I've seen that before.

Comment (jweiser) [reply src/test/resources/log4j2.xml]: Yeah, I think I picked up that way of doing it by watching React tutorials that uses a kind of embedded HTML in JavaScript called JSX.  

The other way people look to be doing it is to have the ">" on the same line as the last attribute:
https://stackoverflow.com/questions/14791592/break-line-on-each-tag-attribute-and-keep-them-aligned-in-visual-studio-html-cod

I can do it that way from now on if you prefer.

Comment (jweiser) [reply src/test/resources/log4j2.xml]: Okay, sounds good.  Here's an example on the JSX attribute formatting: https://github.com/prettier/prettier/issues/5501.

**Pull request merged**